### PR TITLE
Add docker + CI

### PR DESCRIPTION
**Pull request by mrjvs**
changes:
 - update dockerfile with the standardized dockerfile by matthew
 - Added CI that builds docker and pushes the image to the container registry
 - Fix npm build scripts - it just didnt work
 - Added a config.ts that parses the config.json file, so it doesnt require it to be present at compile time.

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+interface Config {
+    bot_token: string;
+    json_db_path: string;
+    sequelize: {
+        force: boolean;
+        alter: boolean;
+    }
+}
+
+const rawConfigData = readFileSync(path.join(process.cwd(), './config.json'), { encoding: 'utf-8' });
```
Comment (jonbarrow) [inline src/config.ts:13]: Spaces -> tabs and a small styling change just to keep it inline with the rest of the codebase

```suggestion
interface Config {
	bot_token: string;
	json_db_path: string;
	sequelize: {
		force: boolean;
		alter: boolean;
	}
}

const rawConfigData = readFileSync(path.join(process.cwd(), './config.json'), {
	encoding: 'utf-8'
});
```

Comment (mrjvs) [reply src/config.ts]: Fixed in latest commit, I thought it had an editorconfig so I didnt pay attention to it

Comment (jonbarrow) [reply src/config.ts]: That took care of the spaces but not the `rawConfigData` formatting

Comment (mrjvs) [reply src/config.ts]: Whoops, didnt notice that changed as well, fixed (I hope?)

**Comment by jonbarrow** [comment]
LGTM

**Pull request merged**